src/classes/RD2_DataMigrationMapper.cls

```diff
@@ -266,7 +266,7 @@ public class RD2_DataMigrationMapper {
             installmentDate = rd.npe03__Date_Established__c;
         }
 
-        return installmentDate.day().format();
+        return installmentDate == null ? null : installmentDate.day().format();
```

Comment:
Is there other logic that will log an exception if this is an Active and Monthly RD and it's unable to determine a DayOfMonth?

Reply:
To check if it is Active, Monthly and both Date Established and Next Donation Date are empty?

Reply:
What I mean is if this is an Active RD and the InstallmentPeriod is Monthly we would normally require a value in DayOfMonth field. If this method now returns a null value to store into the DayOfMonth field what will happen during data migration?  Will the normal trigger validation throw an error that will get logged or would it just allow the change - which would end up leaving the data in invalid state?

Reply:
If there is an Active RD, either Date Established or Next Donation Date need to be provided. If the "Always Use Last Day Of Month" is unchecked, "Day of Month" will be determined during migration correctly.

I added two new tests to verify "Effective Date", "Next Donation Date" and "Day of Month" when either date is missing, the RD has no Opportunities and the "Effective Date" is empty to start with. The "Effective Date" is determined on earliest Opportunity Close Date, or set to Date Established if no Opp exists.  See `shouldConvertOpenRDWhen*()` methods.